rotkehlchen/chain/ethereum/modules/cowswap/decoder.py

```diff
@@ -0,0 +1,253 @@
+                    else:  # Refund
+                        event.event_subtype = HistoryEventSubType.REFUND
+                        event.notes = f'Refund {event.balance.amount} unused ETH from cowswap'
+
+        return None, []
+
+    # --- Aggregator methods ---
+
+    def _find_trades(self, all_logs: list[EvmTxReceiptLog]) -> list[SwapData]:
+        """
+        Finds the emitted Trade events and decodes them.
+        Also handles special cases when ETH is swapped.
+
+        Returns a list of swap datas. Each entry in the list contains basic information about a
+        swap made in the transaction.
+        """
+        trades = []
+
+        for tx_log in all_logs:
+            if tx_log.topics[0] != TRADE_SIGNATURE:
+                continue
+
+            owner_address = hex_or_bytes_to_address(tx_log.topics[1])
+            from_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+            to_token_address = hex_or_bytes_to_address(tx_log.data[32:64])
+            raw_from_amount = hex_or_bytes_to_int(tx_log.data[64:96])
+            raw_to_amount = hex_or_bytes_to_int(tx_log.data[96:128])
+
+            if (
+                from_token_address == A_WETH.resolve_to_evm_token().evm_address and
+                owner_address == ETH_FLOW_ADDRESS
+            ):
+                from_asset = self.eth  # Is swaps are made by eth flow contract in cowswap
+            else:
+                from_asset = EvmToken(evm_address_to_identifier(
+                    address=from_token_address,
+                    chain_id=ChainID.ETHEREUM,
+                    token_type=EvmTokenKind.ERC20,
+                ))
+            to_asset = self.eth if to_token_address == ETH_SPECIAL_ADDRESS else EvmToken(
+                evm_address_to_identifier(
+                    address=to_token_address,
+                    chain_id=ChainID.ETHEREUM,
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            )
+            from_amount = asset_normalized_value(amount=raw_from_amount, asset=from_asset)
+            to_amount = asset_normalized_value(amount=raw_to_amount, asset=to_asset)
+
+            trades.append(SwapData(
+                from_asset=from_asset,
+                from_amount=from_amount,
+                to_asset=to_asset,
+                to_amount=to_amount,
+            ))
+
+        return trades
+
+    def _detect_relevant_trades(
+            self,
+            transaction: EvmTransaction,
+            all_swap_data: list[SwapData],
+            decoded_events: list[HistoryBaseEntry],
+    ) -> list[tuple[HistoryBaseEntry, HistoryBaseEntry]]:
+        """
+        This function does the following
+        1. Detect trades that are relevant to the tracked accounts.
+        2. For each trade, find the corresponding spend event and receive event. For swaps where
+            ETH is spent creates a new ETH spend event.
```

Comment:
Looks in Github as if it's not aligned properly. If it's just Github's interface ignore the comment.

```suggestion
           ETH is spent creates a new ETH spend event.
```